apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -454,6 +453,27 @@ function handleChatCompletionError(
         timestamp: Date.now(),
       })
     );
+  } else if (error instanceof NetworkError && error.response.status === 403) {
+    const responseBody = await error.response.json();
+    const userType = responseBody?.user_type;
+
+    let messageText;
+    if (userType === 'teacher') {
+      messageText = commonI18n.aiChatNotAuthorizedTeacher();
+    } else if (userType === 'student') {
+      messageText = commonI18n.aiChatNotAuthorizedStudent();
+    } else {
+      messageText = commonI18n.aiChatNotAuthorizedSignedOut();
+    }
```

Comment:
This is fine as is, I'm just gonna wax philosophical about how much I hate that javascript makes us define these variables outside of the `if/else` ladder _and_ make them `let` just so we can assign a single thing to them.

So. What alternatives are there? We could create a function to do the return:

```
const getMessageText = (userType) => {
  // arbitrarily make it a switch instead of an if/else ladder, since the logic is simple
  switch(userType) {
    case 'teacher':
      return commonI18n.aiChatNotAuthorizedTeacher();
    case 'student':
      return commonI18n.aiChatNotAuthorizedStudent();
    default:
      return commonI18n.aiChatNotAuthorizedSignedOut();
  }
}
...
const messageText = getMessageText(userType)
```
But that requires a separate function elsewhere and moves the logic. I'm not a big fan.

We could take that function an make it an IIFE.

```
const messageText = ((userType) => {
  switch(userType) {
    case 'teacher':
      return commonI18n.aiChatNotAuthorizedTeacher();
    case 'student':
      return commonI18n.aiChatNotAuthorizedStudent();
    default:
      return commonI18n.aiChatNotAuthorizedSignedOut();
  }
})(userType)
```

That's kinda better in that it keeps the logic in place, and it's probably the most common way to do it. I'm just not a big fan of IIFEs. It feels like a hack to me, so I try to avoid them. Again, totally personal preference.

Instead, I might be likely to use a lookup table with a default:
```
const userTypeToMessageText = {
  'teacher' : commonI18n.aiChatNotAuthorizedTeacher(),
  'student' : commonI18n.aiChatNotAuthorizedStudent()
}

const messageText = useTypeToMessageText[userType] || commonI18n.aiChatNotAuthorizedSignedOut();
```

I tend to like dispatch table style objects because it makes it super clear when you're adding new cases, and you don't need all the extra verbiage from an if/else ladder or even a switch, nor the machinery of an IIFE or an external function.

At the same time, I don't think there's any real benefit to changing it here since it's only two cases. If there were a half a dozen or more, then I think we could get some more clarity with a different approach, but as is? Meh.

Mostly I'm just griping that javascript doesn't have a better built in construct for this. If a `switch` statement just returned a value that we could use in an assignment, all would be right in the world.